pkg/helpers/conditions/conditions.go

```diff
@@ -28,13 +31,28 @@ func PodContainerRunning(containerName string) watchtools.ConditionFunc {
 		case *corev1.Pod:
 			switch t.Status.Phase {
 			case corev1.PodRunning, corev1.PodPending:
-				if t.Status.ContainerStatuses[0].State.Waiting != nil {
-					if t.Status.ContainerStatuses[0].State.Waiting.Reason == "CreateContainerError" {
-						klog.V(0).Info(t.Status.ContainerStatuses[0].State.Waiting.Message)
-						return false, fmt.Errorf(t.Status.ContainerStatuses[0].State.Waiting.Message)
+				for _, s := range t.Status.ContainerStatuses {
+					if s.State.Waiting != nil {
+						if s.State.Waiting.Reason == "CreateContainerError" {
+							klog.V(0).Info(s.State.Waiting.Message)
+							return false, fmt.Errorf(s.State.Waiting.Message)
+						}
 					}
 				}
 			case corev1.PodFailed, corev1.PodSucceeded:
+				for _, s := range t.Status.ContainerStatuses {
+					if s.State.Terminated != nil {
+						exitCode := s.State.Terminated.ExitCode
+						if exitCode != 0 {
+							klog.V(0).Infof("non-zero exit code(%d) from container %s", exitCode, s.Name)
+							// delete the existing pod
+							if err := coreClient.Pods(t.Namespace).Delete(t.Name, metav1.NewDeleteOptions(0)); err != nil && !errors.IsNotFound(err) {
+								return false, fmt.Errorf("unable to delete existing debug pod %q: %v", s.Name, err)
+							}
+							os.Exit(int(exitCode))
```

Comment:
Helpers can't and should never `os.Exit` it should return error and only there should be a proper interpretation of the error and `os.Exit`

Reply:
that was from a previous commit, it was removed in subsequent commit - i'll try to clear the history there.